Improve build deletion query performance

williamjallen opened the pull request:
#1432 added a suboptimal query which is too slow on production systems.  This PR works around the slow part of the query by using far more restrictive conditions, which should improve performance significantly.

zackgalbreath reviewed (approved):
@williamjallen if the commit I added for `coveragefile` looks good to you then I think we're ready to merge this PR.

williamjallen commented:
> @williamjallen if the commit I added for `coveragefile` looks good to you then I think we're ready to merge this PR.

Yep, that looks great.  I missed that the second query was almost identical to the one I refactored.
